src/main.zig

```diff
@@ -838,38 +837,37 @@ pub const KeyCallbackMachine = struct {
     }
 };
 pub const HistoryEntry = struct {
-    entry: ArrayList(u8),
```

Comment:
Storing bytes in a mutable ArrayList just to avoid thinking about cleanup is nuts :P

Really this should be "caller owns memory" (given it's an entry) and then `deinit()` receives an allocator as unmanaged containers usually do to free the slice.

Reply:
I'm still not sold on the unmanaged stuff, but I'll defer to your expertise anyhow.